src/libsystemd/sd-resolve/sd-resolve.c

```diff
@@ -239,14 +239,13 @@ static int send_addrinfo_reply(
 
         assert(out_fd >= 0);
 
-        resp = (AddrInfoResponse) {
-                .header.type = RESPONSE_ADDRINFO,
-                .header.id = id,
-                .header.length = sizeof(AddrInfoResponse),
-                .ret = ret,
-                ._errno = _errno,
-                ._h_errno = _h_errno,
-        };
+        /* Do not use structured initializer here. MSan or valgrind do not like it. */
+        resp.header.type = RESPONSE_ADDRINFO;
+        resp.header.id = id;
+        resp.header.length = sizeof(AddrInfoResponse);
+        resp.ret = ret;
+        resp._errno = _errno;
+        resp._h_errno = _h_errno;
```

Comment:
does this actually suffice? I mean, the padding is still not iniitalized with this change, no?

Reply:
`resp` or `req` are already initialized with zero when they are declared. See b127bc99d13cede3b1ba6bd8acd936ff7a0d2f31.